runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/WindmillSink.java

```diff
@@ -146,8 +146,15 @@ private <EncodeT> ByteString encode(Coder<EncodeT> coder, EncodeT object) throws
           stream.size() == 0,
           "Expected output stream to be empty but had %s",
           stream.toByteString());
-      coder.encode(object, stream, Coder.Context.OUTER);
-      return stream.toByteStringAndReset();
+      ByteString result;
+      try {
+        coder.encode(object, stream, Coder.Context.OUTER);
+        result = stream.toByteStringAndReset();
```

Comment:
nit: I think you can just 
`return stream.toByteStringAndReset();`
and remove the result variable